.github/workflows/validate.yml

```
@@ -0,0 +1,47 @@
+name: Validate newly added JSON
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+jobs:
+  validate-json:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          # Needed to push changes back to the repo
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "22"
+          cache: "yarn"
+          cache-dependency-path: actions/yarn.lock
+
+      - name: Install Dependencies
+        run: yarn install --frozen-lockfile
+        working-directory: ./actions
+
+      - name: Find newly added JSON files
+        id: find-json
+        run: |
+          # Get the list of added JSON files in the records/new/ directory
+          ADDED_FILES=$(git diff ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} --diff-filter=A --name-only | grep '^records/new/.*\.json$')
```

```suggestion
          ADDED_FILES=$(git show HEAD --format= --diff-filter=A --name-only | grep '^records/new/.*\.json$')
```

---

The two are different in case there are multiple commits and they aren't squashed.

---

Right, and in the context of a PR, we're always on a merge commit between the head and base branch.
```suggestion
          ADDED_FILES=$(git diff HEAD^..HEAD --diff-filter=A --name-only records/new | grep '\.json$')
```